static/js/components/EarthquakeMap.jsx

```diff
@@ -0,0 +1,206 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useDispatch } from "react-redux";
+import {
+  ComposableMap,
+  Geographies,
+  Geography,
+  Marker,
+  useZoomPan,
+} from "react-simple-maps";
+import CircularProgress from "@mui/material/CircularProgress";
+import world_map from "../../images/maps/world-110m.json";
+
+let dispatch;
+const width = 700;
+const height = 475;
+
+function CustomZoomableGroup({ children, ...restProps }) {
+  const { mapRef, transformString, position } = useZoomPan(restProps);
+  return (
+    <g ref={mapRef}>
+      <rect width={width} height={height} fill="transparent" />
+      <g transform={transformString}>{children(position)}</g>
+    </g>
+  );
+}
+function setCurrentEarthquakes(currentEarthquakes) {
+  const currentEarthquakeMenu = "Earthquake List";
+  dispatch({
+    type: "skyportal/CURRENT_EARTHQUAKES_AND_MENU",
+    data: { currentEarthquakes, currentEarthquakeMenu },
+  });
+}
+
+function earthquakelabel(nestedEarthquake) {
+  return nestedEarthquake.earthquakes
+    .map((earthquake) => earthquake.event_id)
+    .join(" / ");
+}
+
+function earthquakeCanObserve(nestedEarthquake) {
+  let color = "#f9d71c";
+  if (nestedEarthquake.is_night_astronomical_at_least_one) {
+    color = "#0c1445";
+  }
+  return color;
+}
```

Comment:
Do we care when it is night or day for earthquakes?

Reply:
Good point, will check for retracted or not for now.